Mathlib/Algebra/Ring/Divisibility/Lemmas.lean

```diff
@@ -91,3 +92,32 @@ lemma add_pow_dvd_pow_of_pow_eq_zero_left (hy : y ^ n = 0) :
 end Ring
 
 end Commute
+section CommRing
+
+variable [CommRing R]
```

Comment:
Stupid question: are the two lemma below false in the non-commutative setting?

Reply:
TBH I don't know, the proof surely doesn't work (in practice you write introduce `k1` and `k2`, write the two relation as a matrix multiplication, multiply by the adjugate matrix and the determinant does its magic: this only works over commutative rings).